daemon/api_notices.go

```diff
@@ -124,5 +192,19 @@ func getNotice(c *Command, r *http.Request, user *auth.UserState) Response {
 	if notice == nil {
 		return NotFound("cannot find notice with id %q", noticeID)
 	}
+	if !noticeViewableByUser(notice, requestUID) {
+		return Forbidden("not allowed to access notice with id %q", noticeID)
+	}
 	return SyncResponse(notice)
 }
+
+func noticeViewableByUser(notice *state.Notice, requestUID uint32) bool {
```

Comment:
we probably want a comment somewhere about the fact that snapd has authenticated admins that are not root (this is category that doesn't match pebble way of doing things), and atm we don't have a level of notice visibility that is for all those admins but not other users. We probably don't need such a level but it would be good to capture this, because somebody might be puzzled about that looking at the code.